src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -977,6 +973,7 @@
   <Node Name="BoundBlock" Base="BoundStatementList">
     <Field Name="Locals" Type="ImmutableArray&lt;LocalSymbol&gt;"/>
     <Field Name="LocalFunctions" Type="ImmutableArray&lt;LocalFunctionSymbol&gt;"/>
+    <Field Name="InUnsafeRegion" Type="ThreeState" Null="NotApplicable"/>
```

Comment:
Wouldn't a `bool` be simpler? 
It could represent whether the block had an `unsafe` modifier (`hasUnsafeModifier`).
Then I think only the `new BoundBlock` in `FinishBindBlockParts` would have to be updated (to capture the presence of modifier) and then `RefSafetyAnalysis.VisitBlock(...)` can merge it's `_inUnsafeRegion` with the block's `hasUnsafeModifier` to compute the new `_inUnsafeRegion`.

Reply:
It's not clear how to capture that the block has an `unsafe` modifier. Should `FinishBindBlockParts()` check `node.Parent.Kind() == SyntaxKind.UnsafeStatement"?

Reply:
That would work, but we could also pass a flag from `BindUnsafeStatement` to `BindEmbeddedBlock` then `BindBlock` and `FinishBindBlockParts`.

Reply:
From our discussion, maybe checking the parent node is still the lesser turd :-P